### ISSUE TITLE
Unreachable reached

### ISSUE THREAD

**Issue by denglerchr** [JuliaLang/julia#32131]
Hello all,
I reached the unreachable, in both the new Julia v1.1.1 on Linux and Julia v1.1.0 on Windows. I'm not sure if this is related to https://github.com/JuliaLang/julia/issues/25430, if so, please delete this. Here is the reduced code, producing the error:

```julia
struct Foo{T}
    M::Array{T, 2}
end

function dostuff!(x, u::T, foo::Foo{T}) where {T}
    Msym = Symmetric(foo.M, :U)
    return 1.0
end

# Run example
myfoo = Foo(Array{Float64}(I, 3, 3))
myu = 1.0

test(x) = dostuff!(x, myu, myfoo)

# "dostuff!(1, myu, myfoo)" works for me
test(1)
```

And here the error message (from the Windows machine):
```
Unreachable reached at 000000001F502EDC

Please submit a bug report with steps to reproduce this fault, and any error messages that follow (in their entirety). Thanks.
Exception: EXCEPTION_ILLEGAL_INSTRUCTION at 0x1f502edc -- test at C:\Users\gu62lax\Dropbox\Uni\Projekte\Modules_Julia_1.0\myJLmodules\ChainDyn\examples\crash_minimal.jl:14
in expression starting at C:\Users\gu62lax\Dropbox\Uni\Projekte\Modules_Julia_1.0\myJLmodules\ChainDyn\examples\crash_minimal.jl:17
test at C:\Users\gu62lax\Dropbox\Uni\Projekte\Modules_Julia_1.0\myJLmodules\ChainDyn\examples\crash_minimal.jl:14
unknown function (ip: 000000001F502EFA)
jl_fptr_trampoline at /home/Administrator/buildbot/worker/package_win64/build/src\gf.c:1864
jl_apply_generic at /home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2219
do_call at /home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:323
eval_value at /home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:411
eval_stmt_value at /home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:362 [inlined]
eval_body at /home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:755
jl_interpret_toplevel_thunk_callback at /home/Administrator/buildbot/worker/package_win64/build/src\interpreter.c:885
unknown function (ip: FFFFFFFFFFFFFFFE)
unknown function (ip: 0000000006BD887F)
unknown function (ip: FFFFFFFFFFFFFFFF)
jl_toplevel_eval_flex at /home/Administrator/buildbot/worker/package_win64/build/src\toplevel.c:764
jl_parse_eval_all at /home/Administrator/buildbot/worker/package_win64/build/src\ast.c:883
jl_load at /home/Administrator/buildbot/worker/package_win64/build/src\toplevel.c:826 [inlined]
jl_load_ at /home/Administrator/buildbot/worker/package_win64/build/src\toplevel.c:833
include at .\boot.jl:326 [inlined]
include_relative at .\loading.jl:1038
include at .\sysimg.jl:29
jl_apply_generic at /home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2219
exec_options at .\client.jl:267
_start at .\client.jl:436
jl_apply_generic at /home/Administrator/buildbot/worker/package_win64/build/src\gf.c:2219
jl_apply at /home/Administrator/buildbot/worker/package_win64/build/src\julia.h:1571 [inlined]
true_main at /home/Administrator/buildbot/worker/package_win64/build/ui\repl.c:96
wmain at /home/Administrator/buildbot/worker/package_win64/build/ui\repl.c:217
__tmainCRTStartup at /usr/src/debug/mingw64-x86_64-runtime-5.0.3-1/crt\crtexe.c:329
mainCRTStartup at /usr/src/debug/mingw64-x86_64-runtime-5.0.3-1/crt\crtexe.c:212
BaseThreadInitThunk at C:\windows\System32\KERNEL32.DLL (unknown line)
RtlUserThreadStart at C:\windows\SYSTEM32\ntdll.dll (unknown line)
Allocations: 314565 (Pool: 314485; Big: 80); GC: 0
```

**Comment by StefanKarpinski** [comment]
Thank you for the report and the lovely reduced code to produce it!

**Comment by JeffBezanson** [comment]
Works for me on master. We have fixed quite a few issues like this since 1.1.
